src/query/plan/operator.cpp

```diff
@@ -1258,9 +1290,15 @@ class STShortestPathCursor : public query::plan::Cursor {
       // endpoint we pass to `should_expand`, because everything is
       // reversed.
       for (const auto &vertex : sink_frontier) {
+        if (context.auth_checker &&
+            !context.auth_checker->Accept(context.user, *context.db_accessor, vertex, storage::View::OLD))
+          continue;
+
         if (self_.common_.direction != EdgeAtom::Direction::OUT) {
           auto out_edges = UnwrapEdgesResult(vertex.OutEdges(storage::View::OLD, self_.common_.edge_types));
           for (const auto &edge : out_edges) {
+            if (context.auth_checker && !context.auth_checker->Accept(context.user, *context.db_accessor, edge))
+              continue;
```

Comment:
Something doesn't feel right here
I think source and sink frontier are first {start_vertex}, and {and_vertex}. Which means, they are already processed in scan all. Which means, we should only check the edge, and then the vertex the edge is expanding to. And then we will have it correct through the whole algorithm?